src/question_tree.rb

```diff
@@ -0,0 +1,40 @@
+
+# frozen_string_literal: true
+
+require 'rubytree'
+
+module Metalware
+  class QuestionTree < Tree::TreeNode
```

Comment:
Yeah, I agree with you on this point. The catch being there is a bit of existing code that expects `QuestionTree` to behave like `Tree::TreeNode`. So in the interest of expedience I sub-classed the method to prevent having to change the rest of metalware.

The biggest issue is the question data (`identifiers`, `type`, `default`, etc) is stored within the `content` of the `Tree::TreeNode`. However now there is a `QuestionTree` object, they can be stored as class variables instead. This removes the need to go `question.content.default` for example.

Also in the `Configurator` there is a bit of code to set the `type` if it otherwise hasn't been set. This as well as some of the other `Configurator` code is probably better suited in the `QuestionTree` object as well.

So at this point in time, there is room for improvement on how `QuestionTree` is used. For now this is a starting point in which we can expand in the future.

I'll create a card to address these issues.

Reply:
Your suggestions for improvement sound good, and thanks for creating the card.

For

> The catch being there is a bit of existing code that expects `QuestionTree` to behave like `Tree::TreeNode`. So in the interest of expedience I sub-classed the `QuestionTree` to prevent having to change the rest of metalware.

in particular: often an instance variable which is `delegate`d to as needed can be used to fix this sort of issue; this may or may not help here.

Reply:
Whilst `delegate` would fix that particular issue, a more permanent solution is not to store it in the content at all.

Also there are a few places where the old traversal methods have been used and then the non-questions have been manually filtered out. These cases should be replaced with the new `filtered` traversal methods.

Reply:
I'm not sure how removing the inheritance will affect the traversals, but we can cross that bridge when we get to it.